tools/autograd/gen_variable_type.py

```diff
@@ -117,6 +132,31 @@
 CALL_VIA_DERIVED = CodeTemplate("""\
 baseType->${method_prefix_derived}${base_name}(${unpacked_args})""")
 
+SAVE_TENSOR_STORAGE_PTR = CodeTemplate("""\
+StorageImpl* ${tensor_name}_storage_ptr_saved = (${tensor_name}.defined() && !${tensor_name}.is_sparse()) ? ${tensor_name}.storage().unsafeGetStorageImpl() : nullptr;
```

Comment:
I found that holding onto `Storage` instead of `StorageImpl*` also works (it correctly triggers when we don't have the fix for `gesv`). I fixed it in the new commits.